### PR TITLE
Better SQL query splitter

### PR DESCRIPTION
**Pull request by rodrigozhou**
**What changed?**
Implemented a better SQL query splitter. It splits the statements taking into consideration `BEGIN/END` blocks.

**Why?**
The current algorithm simply splits by semicolon (`;`), which would lead to wrong query statements when there are more complex statements that contains `BEGIN/END` blocks (eg: when creating a trigger).

**How did you test it?**
Wrote unit tests.

**Potential risks**
No issues, it parses the existing SQL files correctly.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: common/persistence/query_util.go [hunk 1]
```diff
@@ -57,22 +66,106 @@ func LoadAndSplitQuery(
...
 	return result, nil
 }
+
+// matchWord is a simple test to check if it matches the whole word:
+// it checks if the adjacent charactes are not letters if they exist.
+func matchWord(s, word string, pos int) bool {
+	if pos+len(word) > len(s) || s[pos:pos+len(word)] != word {
+		return false
+	}
+	if pos > 0 && unicode.IsLetter(rune(s[pos-1])) {
```
Comment (yux0) [inline common/persistence/query_util.go:164]: What about '7BEGIN'? Is it valid?

Comment (rodrigozhou) [reply common/persistence/query_util.go:164]: Not really, I can add this check as well. I wasn't trying to be comprehensive in those checks though. Probably a lot of other edge cases not handled, but it should cover the basic cases, and the rest, the DB will fail the query if it's invalid.

**Comment by MichaelSnowden** [comment]
Could we use a library for this? The code here seems pretty complicated.

**Comment by rodrigozhou** [comment]
@MichaelSnowden I did a quick search before, and I didn't find anything for our use case.

File: common/persistence/query_util_test.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	s.logger = log.NewTestLogger()
+	// Have to define our overridden assertions in the test setup. If we did it earlier, s.T() will return nil
+	s.Assertions = require.New(s.T())
+}
+
+func (s *queryUtilSuite) TestLoadAndSplitQueryFromReaders() {
+	input := `
+		CREATE TABLE test (
+			id BIGINT not null,
+			col1 BIGINT, -- comment with unmatched parenthesis )
```
Comment (alexshtin) [inline common/persistence/query_util_test.go:63]: Add
```
--END
```
and 
```
--begin
```
somewhere too, please.

**Review by alexshtin — APPROVED** [review]

File: tools/common/schema/util.go [hunk 3]
```diff
@@ -37,43 +34,5 @@ const newLineDelim = '\n'
 // and returns an array of cql / sql statements on
 // success.
 func ParseFile(filePath string) ([]string, error) {
```
[the remark this answers is not in the document]
Comment (rodrigozhou) [reply tools/common/schema/util.go]: Removed file, and updated references.

File: common/persistence/query_util.go [hunk 4]
```diff
@@ -53,26 +64,121 @@ func LoadAndSplitQuery(
...
 			result = append(result, stmt)
 		}
-
 	}
 	return result, nil
 }
+
+// matchWord is a simple test to check if it matches the whole word:
+// it checks if the adjacent charactes are not letters if they exist.
+func matchWord(s, word string, pos int) bool {
```
[the remark this answers is not in the document]
Comment (rodrigozhou) [reply common/persistence/query_util.go]: Updated

File: common/persistence/query_util_test.go (hunk 2 above)
Comment (rodrigozhou) [reply common/persistence/query_util_test.go:63]: Updated

**Pull request merged**